### PR TITLE
[Delivers #90634128] Error message when approving your own cart

### PR DESCRIPTION
**Pull request by nickbristow**
Addresses https://www.pivotaltracker.com/story/show/90634128

### CODE REVIEW

File: app/controllers/communicarts_controller.rb [hunk 1]
```diff
@@ -21,10 +21,12 @@ def send_cart
   def approval_response
     cart = Cart.find(params[:cart_id]).decorate
     client_data = cart.proposal.client_data_legacy
-    approval = cart.approvals.find_by(user_id: user_id) 
+    approval = cart.approvals.find_by(user_id: user_id)
```
Comment (afeld) [inline app/controllers/communicarts_controller.rb]: Let's use `cart.approvals.approvable.find_by(user_id: user_id)` instead of the separate check below.

Comment (afeld) [inline app/controllers/communicarts_controller.rb]: Though I suppose that wouldn't allow the error message for that specific case to be shown...though not sure that needs to handled separately from a "you aren't allowed to approve this" message.

**Comment by cmc333333** [comment]
The so-called "raphy" case. Mind merging master? I defer to @afeld re `is_approvable?` vs `approvable?`

**Comment by afeld** [comment]
> `is_approvable?` vs `approvable?`

The latter is more Ruby-ish.

File: app/controllers/communicarts_controller.rb [hunk 2]
```diff
@@ -21,9 +21,16 @@ def send_cart
   def approval_response
     cart = Cart.find(params[:cart_id]).decorate
     client_data = cart.proposal.client_data_legacy
-    approval = cart.approvals.find_by(user_id: user_id)
+    approval = cart.approvals.approvable.find_by(user_id: user_id)
+    
+    if !approval
+      flash[:error] = "Sorry. You are not allowed to approve your own request."
```
Comment (afeld) [inline app/controllers/communicarts_controller.rb]: Trying to think if there are other cases where the `approval` could be `nil`...if we want to show this message specifically, might be better to go back to the way you were handling it before. Sorry!

File: app/controllers/communicarts_controller.rb [hunk 3]
```diff
@@ -21,9 +21,16 @@ def send_cart
...
+      redirect_to cart_path(cart)
+      return
```
Comment (afeld) [inline app/controllers/communicarts_controller.rb]: I'm personally allergic to mid-method `return`s...mind using an `if`/`else` instead? @cmc333333 Curious to hear your thoughts.

**Pull request merged**